nova/virt/lxd/driver.py

```diff
@@ -1135,8 +1135,9 @@ def _generate_profile_data(
         vcpus = instance.flavor.vcpus
         if vcpus >= 0:
             config['limits.cpu'] = str(vcpus)
-        config['raw.lxc'] = 'lxc.console.logfile={}\n'.format(
-            instance_attributes.console_path)
+        config['raw.lxc'] = '\n'.join([
+            'lxc.console.logfile={}'.format(instance_attributes.console_path),
+            'lxc.environment.CLOUD_PLATFORM=OpenStack Nova']) + '\n'
```

Comment:
"raw.lxc" should be avoided whenever possible, and this is a situation where LXD does offer the feature in its own configuration.